amy/autoemails/actions.py

```diff
@@ -1235,3 +1244,105 @@ def get_merge_data(self) -> Optional[Dict]:
+    See amy.autoemails.management.commands.repeated_jobs
+    """
+
+    launch_at = timedelta(hours=1)
+
+    def recipients(self) -> Optional[Tuple[str]]:
+        """Assuming self.context is ready, overwrite email's recipients
+        with selected ones."""
+        try:
+            person_email = self.context_objects["person_email"]
+            return (person_email,)
+        except (KeyError, AttributeError):
+            return None
+
+    def all_recipients(self) -> str:
+        """If available, return string of all recipients."""
+        recipients = self.recipients()
+        if recipients is not None:
+            return ", ".join(recipients)
+        return ""
+
+    def reply_to(self) -> Optional[Tuple[str]]:
+        """Overwrite in order to set own reply-to from descending Action."""
+        return (settings.ADMIN_NOTIFICATION_CRITERIA_DEFAULT,)
+
+
+class BaseRepeatedAction:
+    """
+    Base class for repeated jobs in the Redis.
+    """
+
+    trigger: Trigger
+    launch_at: Optional[timedelta] = None
+    EMAIL_ACTION_CLASS: BaseAction
+    INTERVAL: int = DAY_IN_SECONDS  # time between repeats
+    REPEAT: Optional[int] = None  # number of times the job is repeated
+
+    def __init__(self, trigger: Trigger):
+        super().__init__()
+        self.trigger = trigger
+
+    def __call__(self, *args, **kwargs):
+        raise NotImplementedError
+
+    def get_launch_at(self, *args, **kwargs) -> Optional[timedelta]:
+        return self.launch_at
+
+    @property
+    def template(self):
+        return self.trigger.template
+
+
+class UpdateProfileReminderRepeatedAction(BaseRepeatedAction):
+    """
+    Daily job that runs determining what emails to send out to people
+    to remind them when they should log in and update their profile.
+    """
+
+    launch_at = timedelta(hours=1)
+    LAST_LOGGED_IN_DELTA = timedelta(days=365)
+    EMAIL_ACTION_CLASS = ProfileUpdateReminderAction
+
+    def __call__(self, *args, **kwargs):
+        people = self.get_people_with_anniversary()
+        triggers = Trigger.objects.filter(
+            active=True,
+            action="profile-update",
+        )
+        logger.debug("TRIGGER %s", triggers)
+        for person in people:
+            ActionManageMixin.add(
+                action_class=self.EMAIL_ACTION_CLASS,
+                logger=logger,
+                scheduler=scheduler,
+                triggers=triggers,
+                context_objects={
+                    "person_email": person.email,
+                },
+                object_=person,
+            )
+
+    @staticmethod
+    def get_people_with_anniversary() -> Iterable[Person]:
+        """
+        Pull all people whose created_at anniversary is today
+        """
+        today = datetime.today()
+        people = Person.objects.filter(
+            is_active=True,
+            created_at__month=today.month,
+            created_at__day=today.day,
+        )
```

Comment:
Should we only include people who accepted being contacted?

Reply:
everyone should be contacted here regardless of whether they accepted being contacted as the yearly reminder can be considered a legal notice.